Flabbergasted-Flamingos/bangazon-api
# Jdb permissions

# Description
This PR adds the softdelete module to the model structure to prevent foreign key objects from being deleted accidentally.  There was also an empty class that was duplicating another working class which I removed, along with some commented code that was no longer relevant.  

Finally, on a few modules I limited the http verbs that are allowed by the API.    

## Number of Fixes
3

## Steps to Test Solution
in your virtual environment:
```
pip install django-safedelete
```
```
git fetch --all
git checkout JDB_permissions
rm -rf api/migrations
rm db.sqlite3
python manage.py makemigrations api
python manage.py migrate
```

make some data. check that DELETE is not allowed on: Customers, Departments, Employees.
